include/mbedtls/check_config.h

```diff
@@ -284,7 +284,8 @@
 
 /* Helper for ECDSA dependencies, will be undefined at the end of the file */
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-#if defined(PSA_HAVE_FULL_ECDSA)
+#if defined(PSA_WANT_ALG_ECDSA) && defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR) && \
+    defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY)
```

Comment:
I think the last one is redundant, considering the fact what `WANT_xxx_KEY_PAIR` implied `WANT_xxx_PUBLIC_KEY` as Gilles mentioned.